boot/bootutil/src/loader.c

```diff
@@ -933,21 +933,23 @@ boot_validated_swap_type(struct boot_loader_state *state,
         vtable = (uint32_t *)(vtable_addr);
         reset_addr = vtable[1];
 #ifdef PM_S1_ADDRESS
-        const struct flash_area *primary_fa;
-        int rc = flash_area_open(flash_area_id_from_multi_image_slot(
```

Comment:
This would work for the network image pair, but its primary slot is emulated in RAM, therefore the flash_area doesn't have real offset.